File: rust-client/library/prompt.rs
use anyhow::{Context, Result}; // Importing Result and Context from anyhow crate

pub async fn prompt(prompt: &str, api_key: &String) -> Result<String> {
    let client = reqwest::Client::new();
  
    let request_data = serde_json::json!({
        "model": "gpt-3.5-turbo",
        "messages": [{ "role": "user", "content": prompt }],
        "temperature": 0.7,
    });
  
    let response = client
        .post("https://api.openai.com/v1/chat/completions")
        .header("Content-Type", "application/json")
        .header("Authorization", format!("Bearer {}", api_key))
        .json(&request_data)
        .send()
        .await?
        .text()
        .await?;
    
    let response_data: serde_json::Value = serde_json::from_str(&response)?;
    let code = response_data["choices"][0]["message"]["content"]
        .as_str()
        .ok_or_else(|| anyhow::anyhow!("Response data does not contain expected content"))?;
  
    Ok(code.to_string())
  }

File: library/prompt.ts

export async function prompt(prompt: string, apiKey: string): Promise<string> {
    let retries = 3; // Retry up to 3 times
    while (retries > 0) {
        try {
            const MODEL = {} as any;
            /*const response = await fetch('https://api.openai.com/v1/chat/completions', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                    Authorization: `Bearer ${apiKey}`,
                },
                body: JSON.stringify({
                    model: 'gpt-3.5-turbo',
                    messages: [{ role: 'user', content: prompt}],
                    temperature: 0.7,
                }),
                });*/
            
            let code = '';
            if (MODEL.ok) {
                const responseData = await MODEL.json();
                code = responseData.choices[0].message.content;
            } else {
                console.error('Failed to fetch data:', MODEL.status, MODEL.statusText);
                code = "Failed to fetch data: " + MODEL.status + " " + MODEL.statusText;
            }
            return code;
        } catch (error) {
            console.error('Error fetching data:', error);
            retries--;
        }
    }
    throw new Error('Failed to fetch data after multiple retries');
}
